### ISSUE TITLE
Error in hicDifferentialTAD when using cool

### ISSUE THREAD

**Issue by lldelisle** [deeptools/HiCExplorer#587]
Hi,
Here you assume that you have only one chromosome but this is not the case, it is dependent on the number of thread...
https://github.com/deeptools/HiCExplorer/blob/1fad77274e838a98dbfab591524ee0571203c03c/hicexplorer/hicDifferentialTAD.py#L102-L103
Then it raises an issue:
```
Traceback (most recent call last):
  File "/home/ldelisle/miniconda2/envs/hicexplorer3.5/lib/python3.6/site-packages/hicmatrix/lib/cool.py", line 103, in load
    matrix = cooler_file.matrix(balance=False, sparse=True, as_pixels=False).fetch(self.chrnameList[0]).tocsr()
  File "/home/ldelisle/miniconda2/envs/hicexplorer3.5/lib/python3.6/site-packages/cooler/core.py", line 573, in fetch
    i0, i1, j0, j1 = self._fetch(*args, **kwargs)
  File "/home/ldelisle/miniconda2/envs/hicexplorer3.5/lib/python3.6/site-packages/cooler/api.py", line 384, in _fetch
    region1 = parse_region(region, self._chromsizes)
  File "/home/ldelisle/miniconda2/envs/hicexplorer3.5/lib/python3.6/site-packages/cooler/util.py", line 160, in parse_region
    chrom, start, end = parse_region_string(reg)
  File "/home/ldelisle/miniconda2/envs/hicexplorer3.5/lib/python3.6/site-packages/cooler/util.py", line 136, in parse_region_string
    start, end = _expect(_tokenize(parts[1]))
  File "/home/ldelisle/miniconda2/envs/hicexplorer3.5/lib/python3.6/site-packages/cooler/util.py", line 126, in _expect
    raise ValueError("End coordinate less than start")
ValueError: End coordinate less than start
ERROR:hicmatrix.lib.cool:Error: End coordinate less than start

```
And the main is waiting for ever...

**Comment by joachimwolff** [comment]
I will have a look as soon as I am done with my publications. So maybe in September?!

**Comment by joachimwolff** [comment]
Should be fixed with release 3.5.2. Please reopen if the issue still exists in this release.

**Comment by dawe** [comment]
I'm experiencing the same issue (v3.6) but it seems to be fixed by sorting TAD file by position
